### DISCUSSION TITLE
Welcome to html5-qrcode Discussions!

### DISCUSSION THREAD

**Discussion by mebjas** [mebjas/html5-qrcode#148] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by laikuchan** [comment]
Hi,
Thanks for this!
 Question - is there a way to scan the same QR code twice using Html5QrcodeScanner  without creating a new Html5QrcodeScanner obj?

I'm using:
if (lastResult !== qrCodeMessage)  { //logic here}

But this is no good if the user wants to scan their QR code again

thanks

**Reply by mebjas** [reply]
What is the action taken after user scans a code?

**Comment by lanost** [comment]
Is there a way in which I can get the code type on success. Like code-128 or qr_code? I have a scenario where I need to know which type of code is scanned by the scanner, is it a barcode or a qr code etc.

Without using any framework

**Reply by mebjas** [reply]
Can do, will add, please file a feature request and assign it to me.

**Reply by lanost** [reply]
Thanks for returning the format. Is there a way i can get the captured image URL as well.

**Reply by mebjas** [reply]
What do you mean by captured image URL?

**Reply by lanost** [reply]
The video screenshot at the time of scan, means the barcode or qr code scanned as a local image url

**Reply by lanost** [reply]
@mebjas is this possible ?. My scenario is that I need to show the qr code image which was actually scanned by the scanner along with some details fetched from the db. So that the user can verify the details.

**Reply by lanost** [reply]
@mebjas i found a workaround. posting it here if it helps someone in future.

`function capture(scaleFactor) {
                var video = document.getElementById('qr-reader__scan_region').querySelector('video')
                if (scaleFactor == null) {
                    scaleFactor = 1;
                }
                var w = video.videoWidth * scaleFactor;
                var h = video.videoHeight * scaleFactor;
                var canvas = document.createElement('canvas');
                canvas.width = w;
                canvas.height = h;
                var ctx = canvas.getContext('2d');
                ctx.drawImage(video, 0, 0, w, h);
                return canvas.toDataURL();
            }`

call function like this

`if (document.getElementById('qr-reader__scan_region').querySelector('video')) {
                    scanned_image = capture(0.25);
                } else if ($('#qr-canvas-visible')) {
                    scanned_image = $('#qr-canvas-visible')[0].toDataURL()
                }`

calling this function from onsuccess function will return the screen capture as a data URL. @mebjas if you can provide this in the output itself for both camera scanner and file upload it would be helpful for many scenarios.

**Reply by mebjas** [reply]
This is not inbuilt in the library, check documentation.

**Comment by lanost** [comment]
Scanner not working for HD 720p web camera Logitech. camera window is working just fine but QR code scanning not producing any result. but working just fine with integrated camera.

![Screenshot from 2021-06-15 10-21-46](https://user-images.githubusercontent.com/22130475/121994808-8e9e7c00-cdc3-11eb-8690-b58cc15cdfe7.png)

**Reply by mebjas** [reply]
Looks like it's not able to focus, can you take the camera closer and try to scan.

**Reply by lanost** [reply]
Tried it. when bringing too much closer image becomes blur. still it doesn't work with the webcam

**Reply by mebjas** [reply]
Is other type of QR Code scanner able to scan with the webcam?

**Reply by lanost** [reply]
I tried with another library as well it is also unable to scan the QR code using the web cam. integrated camera works fine there too.

**Comment by ROBERT-MCDOWELL** [comment]
@lanost
(cannot sub comment today, github bug?)
it seems your logitech camera has "white" not really white or maybe with too much dirt into,
also try to have your QrCode in the middle of a white page, not in the corner and see if it's better

**Reply by lanost** [reply]
Thanks for pointing that out. I tried testing with another webcam and it worked.

**Comment by ROBERT-MCDOWELL** [comment]
@lanost
glad you solved it, damn hardware :)

**Comment by FredmanW** [comment]
Hello, Where do I add a feature request? I'd like to have the Micro QR code format included. It is useful for small surface areas. It is as I understand the smallest size QR code format. I think it would be quite valuable to add. Thanks

**Comment by FredmanW** [comment]
Hello, For a feature request, I think ability to show multiply scans would be useful if possible in scanning a group of codes at once and listing all scanned in the group.

**Comment by ROBERT-MCDOWELL** [comment]
well, you can easily program it as it's a very specific feature.

**Reply by FredmanW** [reply]
Thanks, but I'm not a programmer. So would a basic skilled developer understand how to add the Micro QR code format
? And if it is so easy, then why wasn't it added already?  Or maybe you're referring to the group scanning, I also was asking as a feature request?

**Comment by ROBERT-MCDOWELL** [comment]
your feature request is too specific to be developed into the global code.
you should hire a dvelopper for that.

**Comment by ROBERT-MCDOWELL** [comment]
btw I answer to the " I think ability to show multiply scans"

**Reply by FredmanW** [reply]
Thanks, and any thoughts about the Micro QR code feature?

**Comment by ROBERT-MCDOWELL** [comment]
golden rules before to ask a code improvement:
- check if your camera has no focus and color issues
- check if the room luminosity is fair enough for any scan
- check with multiple device with camera and quality to be sure that the issue is reproducing
Then for specific barcodes always provide a sample of which one you are talking about

**Comment by stalker24rus** [comment]
Hi, Minhaz!
Thank you for library.

If I use all your scripts into HTML, all works good. 
But I can not include your library in React component correctly.
Raised exception 'Uncaught TypeError: Cannot read properties of undefined (reading 'Html5Qrcode')'.
What could it be? Help me, please.

P.S. I am new in JS and react...

See image:
<img width="1214" alt="Снимок экрана 2021-09-17 в 00 41 00" src="https://user-images.githubusercontent.com/46667647/133661457-deb499ad-7460-4597-b7b0-6c74a9edb115.png">
<img width="792" alt="Снимок экрана 2021-09-17 в 00 43 44" src="https://user-images.githubusercontent.com/46667647/133661476-10169aad-0147-40ce-ad35-6eef2fe22f14.png">

**Reply by mebjas** [reply]
Yeah this is a known issue, I'll be looking into this.

**Comment by gerf1** [comment]
Hi,

I have a problem that whenever I open the page the camera activates itself and the smaller screen appears with no focus in the middle like this:
![camera2](https://user-images.githubusercontent.com/17769623/154698494-2cd97055-afb7-4377-a8db-c2ca66782029.png)

Is there any way to fix this or directly open this page first without opening the camera at first like this?
![camera](https://user-images.githubusercontent.com/17769623/154698572-0052ac15-8c1c-46aa-9325-90a59427b8b0.png)

**Comment by thrilosgate7** [comment]
Is it possible instead only to read the qr code to open the integreded link from the qr code? If so what have i change from the js file or html page? thx in advance.

**Reply by Mathias89** [reply]
see #106

**Comment by isakovsv** [comment]
Hi, Minhaz!
Thank you for library. Great one!
It works fine for me except qr code contains a lot of data. Is there a limit for this library? If positive How can I scan a fat qr?

Thanks in advance ))
Serge
